src/OrchardCore.Modules/OrchardCore.Deployment.Remote/Views/RemoteClient/Create.cshtml

```diff
@@ -9,14 +9,47 @@
         <span asp-validation-for="ClientName" class="text-danger"></span>
     </div>
 
-    <div class="mb-3" asp-validation-class-for="ApiKey">
-        <label asp-for="ApiKey">@T["Api Key"]</label>
-        <input asp-for="ApiKey" type="password" class="form-control" />
-        <span asp-validation-for="ApiKey" class="text-danger"></span>
+    <div class="mb-3">
+        <div asp-validation-class-for="ApiKeySecret">
+            <label asp-for="ApiKeySecret">@T["Api Key Secret"]</label>
+            @await Component.InvokeAsync(
+            "SelectSecret",
+            new
+            {
+            secretType = "TextSecret",
+            selectedSecret = Model.ApiKeySecret,
+            htmlId = Html.IdFor(m => m.ApiKeySecret),
+            htmlName = Html.NameFor(m => m.ApiKeySecret),
+            required  = false,
+            })
+            <span asp-validation-for="ApiKeySecret"></span>
+            <span class="hint">@T["The api key secret."]</span>
```

Comment:
Explain here and in the other UI too that either this or the plain API key should be used but not both.

Reply:
When a secret is selected the plain text key is no longer rendered unless you select again `None` in the secrets list, so in the UI you can't see a selected secret and a plain text at the same time.

When a secret is selected a plain text may still exist but in the code the secret has an higher precedence.

Reply:
OK!